simplecpp.cpp

```diff
@@ -2341,6 +2341,63 @@ static void simplifySizeof(simplecpp::TokenList &expr, const std::map<std::strin
     }
 }
 
+/** Evaluate __has_include(file) */
+static std::string openHeader(std::ifstream &f, const simplecpp::DUI &dui, const std::string &sourcefile, const std::string &header, bool systemheader);
+static void simplifyHasInclude(simplecpp::TokenList &expr, const simplecpp::DUI &dui)
+{
+    for (simplecpp::Token *tok = expr.front(); tok; tok = tok->next) {
+        if (tok->str() != "__has_include")
+            continue;
+        simplecpp::Token *tok1 = tok->next;
+        if (!tok1) {
+            throw std::runtime_error("missing __has_include argument");
+        }
+        simplecpp::Token *tok2 = tok1->next;
+        if (!tok2) {
+            throw std::runtime_error("missing __has_include argument");
+        }
+        if (tok1->op == '(') {
+            tok1 = tok1->next;
+            while (tok2->op != ')') {
+                tok2 = tok2->next;
```

Comment:
should we abort if there is an inner "(" ?

Reply:
According to https://en.cppreference.com/w/cpp/preprocessor/include, having `)` in filenames is legal, so ideally we should check the next token, and:

- if it starts with `"`, just take it as it is, `)` will be a part of it,
- if it starts with `<`, we shall parse till the next `>` token,

and then the next token should be `)`.

But I feel like having `)` in filenames is a pretty obscure edge case.